src/structures/ApplicationCommand.js

```diff
@@ -0,0 +1,150 @@
+   */
+  get createdTimestamp() {
+    return SnowflakeUtil.deconstruct(this.id).timestamp;
+  }
+
+  /**
+   * The time the command was created at
+   * @type {Date}
+   * @readonly
+   */
+  get createdAt() {
+    return new Date(this.createdTimestamp);
+  }
+
+  /**
+   * Data for creating or editing an application command.
+   * @property {string} name The name of the command
+   * @property {string} description The description of the command
+   * @property {ApplicationCommandOption[]} [options] Options for the command
+   * @typedef {Object} ApplicationCommandData
+   */
+
+  /**
+   * Edits this application command.
+   * @param {ApplicationCommandData} data The data to update the command with
+   * @returns {ApplicationCommand}
+   */
+  async edit(data) {
+    const raw = {};
+    if (data.name) raw.name = data.name;
+    if (data.description) raw.description = data.description;
+    if (data.options) raw.options = data.options.map(this.constructor.transformOption);
+
+    let path = this.client.api.applications(this.client.application.id);
+    if (this.guild) {
+      path = path.guilds(this.guild.id);
+    }
+    const patched = await path.commands(this.id).patch({ data: raw });
+
+    this._patch(patched);
+    return this;
+  }
+
+  /**
+   * Deletes this command.
+   * @returns {ApplicationCommand}
+   */
+  async delete() {
+    let path = this.client.api.applications(this.client.application.id);
+    if (this.guild) {
+      path = path.guilds(this.guild.id);
+    }
+    await path.commands(this.id).delete();
+
+    if (this.guild) {
+      this.guild.commands.cache.delete(this.id);
+    } else {
+      this.client.application.commands.cache.delete(this.id);
+    }
+
+    return this;
+  }
+
+  /**
+   * An option for an application command or subcommand.
+   * @typedef {Object} ApplicationCommandOption
+   * @property {ApplicationCommandOptionType} type The type of the option
+   * @property {string} name The name of the option
+   * @property {string} description The description of the option
+   * @property {boolean} [required] Whether the option is required
+   * @property {ApplicationCommandOptionChoice[]} [choices] The choices of the option for the user to pick from
+   * @property {ApplicationCommandOption[]} [options] Additional options if this option is a subcommand (group)
+   */
+
+  /**
+   * A choice for an application command option.
+   * @typedef {Object} ApplicationCommandOptionChoice
+   * @property {string} name The name of the choice
+   * @property {string|number} value The value of the choice
+   */
+
+  /**
+   * Transforms an {@link ApplicationCommandOption} object into something that can be used with the API.
+   * @param {ApplicationCommandOption} option The option to transform
+   * @returns {Object}
+   * @private
+   */
+  static transformOption(option) {
+    return {
+      ...option,
+      type: ApplicationCommandOptionTypes[option.type],
+      options: option.options?.map(this.transformOption),
```

Comment:
```suggestion
      options: option.options?.map(opt => this.transformOption(opt)),
```
or
```suggestion
      options: option.options?.map(this.transformOption.bind(this)),
```

Reply:
This doesn't seem to be required: [Playground](https://www.typescriptlang.org/play?#code/MYGwhgzhAEByCuBbARgUwE4wN4Chr+gG0ByYAewDsIAXdeYas9YgXQC5pqBPAB1TIBmcJGkwBuPAURgefACYB+DhREZCLCZPzSeACgCU0XAROcAFgEsIAOh3zoAXiIBGADTQATO4DMLWzN1qSxtyKlp6RnRrMDk5ABUAdzJ9CRMAXxwtaBowagtgaBj4pN0VRGVVdENjU3x0VGp4dApoMugAak9UggyMnFBIGDjUGiMsstEIDgQUDAhNE1CaOgYmAzHa-CCrawm5x1bUBOFZ8SzezKXqThHrpwoj6GGaCWpb3cqbHQNNJbIQVDWEBkADmgXee0w-lkqDkKRwQA)

I also just tested creating / fetching application commands, and it worked fine without this change

Reply:
oh nevermind, the error comes when it calls itself recursively, i'll just change it to the class name